dev/archery/archery/docker/tests/test_docker.py

```diff
@@ -252,12 +252,12 @@ def test_arrow_example_validation_passes(arrow_compose_path):
 def test_compose_default_params_and_env(arrow_compose_path):
     compose = DockerCompose(arrow_compose_path, params=dict(
         UBUNTU='18.04',
-        DASK='master'
+        DASK='main'
```

Comment:
Thank you for noticing this and bringing it up!

I attempted to fix this by modifying the install scripts to use Git to compute the current default branch, however, at the time of running the scripts, the working directory is not within a git repository. 

Since the flag value that is used is not dependent on the actual default branch name of the Arrow or Pandas repositories, it could be helpful to change the flag value to `default`, or one of the following:
- `devel`
- `upstream_devel`

If this sounds good to you, I can make this change and also update the documentation page at: https://arrow.apache.org/docs/developers/continuous_integration/docker.html#usage to reflect the new flag value.

Reply:
Your approach sounds good to me, as this name refers to the upstream development version of the related project (pandas, dask) it makes more sense to use a flag like `upstream_devel` instead of `master` or `main` to differentiate this vs the `latest` release, the `nightly` build or a specific version.

Reply:
Thank you @raulcd! I've made this change and qualified by running the sample command from your original comment and running the affected tests in `dev/archery/archery/docker/tests/test_docker.py`.

I've also made the changes to the documentation and qualified by performing a sphinx build.

Reply:
sorry for the late reply. We also have to update the `tasks.yml` definition to make crossbow run the correct task versions, updating `master` for `upstream_devel` on the following:
https://github.com/apache/arrow/blob/master/dev/tasks/tasks.yml#L1495
and
https://github.com/apache/arrow/blob/master/dev/tasks/tasks.yml#L1515